### ISSUE TITLE
Client connection fails with non-alphanumeric characters in password

### ISSUE THREAD

**Issue by glitchbunny** [lutris/lutris#918]
I created a new Lutris account with a randomly-generated password, which worked as expected, leaving me logged in on the web site.

However attempting to subsequently connect using the Lutris client, resulted in login failure, with a HTTP 400 error:

ERROR    2018-06-14 23:07:55,704 [api]:Unable to connect to server (https://lutris.net/api/accounts/token): HTTP Error 400: Bad Request

The generated password contained a backslash.  Guessing that this was the problematic character, I attempted to change it using the still-logged-in session in the web browser, but this also failed to work, due to the old password being unrecognised (can't remember the precise error here, sorry!).

I then used the forgotten password function to change the password (this worked as expected), replacing the backslash character, but the new backslash-less password also suffered from the same login problem in the Lutris client.

So I then changed the password again, replacing all non-alphanumeric characters.  This resulted in a successful connection in the Lutris client.

The replaced characters were: ( % # < -

And of course originally there was a backslash that was also potentially problematic, but at least one of those other characters is too.

**Comment by strycore** [comment]
The backslash plus the character following it was probably the issue, since it's an escape character.

**Comment by sigmaSd** [comment]
Tried again with this password    \a\a   still cant reproduce .

**Comment by glitchbunny** [comment]
I'll set up the dev environment (unless there is a handy test system somewhere?) and try it again, but this time if I can reproduce the problem I'll attempt to dig a little deeper (last time, I was only creating an account for personal use rather than attempting to debug anything).

**Comment by strycore** [comment]
I wonder if this only happens with special escape characters, like /n or /t

**Comment by glitchbunny** [comment]
I wondered that too - the original password that I had trouble with was:

YfUCwf(%#e<64\GZ

**Comment by glitchbunny** [comment]
I have found that changing to the original problematic password in an account that already exists does not cause the problem.

Creating a new account with the problematic password does result in an inability to connect in the Lutris client.  I also found that the inability to connect persists even after the password is changed and the new password is entered in the client, the Lutris client must be restarted for the new password to work.

I also tested creating an account with an alphabetic-only password but I made a typo when I first attempted to connect, correcting the typo still resulted in connection failure until the Lutris client was restarted so I think that there is possibly a second bug here - that connection failure needs a restart to clear even after the original problem is fixed.  Restarting and connecting with the alphanumeric-only password did work.

**Comment by sigmaSd** [comment]
Weird I tested your original pass and pass that has \n\t still cant reproduce , do you use lutris 0.4.18 ?

**Comment by glitchbunny** [comment]
Yes, client version 0.4.18.  On a Debian testing (Buster) machine.  The machine in question was originally installed as Stretch and then sources switched and dist-upgraded.

I have also had a successful connection with that password too now and am starting to suspect that the bug I saw was actually just that connection problems persist and all my password choices and changes were merely incidental.

Would you be able to test the following?;

- Close the Lutris client and then start it again.
- Connect to lutris.net but misspell the password.  Connection should (obviously) fail.
- Now attempt connecting again with the correct password.

Does it work or does the client continue to tell you that the connection failed?

**Comment by sigmaSd** [comment]
followed your steps exactly, still cant reproduce (unfortunately ?) .

**Comment by glitchbunny** [comment]
Agh!  I am now also unable to reproduce, following the steps that had apparently "worked" for me previously in two separate accounts I created for the purpose.  Perhaps the root cause is something else entirely?  Random network glitch my end?  Simply a series of mistakes on my part?  I'm beginning to question the very nature of personal experience!  :)

I'm so sorry for sending you on a wild goose chase.  I think the best thing to do might be to close this as unreproducible and I'll post again if I can determine some reliable way to trigger the behaviour that I observed previously.

**Comment by sigmaSd** [comment]
That's probably what you should do for now.
Its cool mate it would've been an interesting bug ^^
